view/frontend/templates/js/replacejs.phtml

```diff
@@ -153,590 +150,687 @@ window.boltConfig = <?php echo $block->getSettings(); ?>;
-    onElementReady(bolt_button_selector, function(element) {
-        checkout_buttons = true;
-        processButtons();
-    });
-    ////////////////////////////////////////////////
-
-    //////////////////////////////////////////////////////////
-    // loop through selectors array and set up the replacement
-    //////////////////////////////////////////////////////////
-    for (var i = 0, length = settings.selectors.length; i < length; i++) {
-        var selector = settings.selectors[i];
-        ! function(selector) {
-            var parts = selector.split('|');
-            // the CSS selector
-            var identifier = parts[0].trim();
-            // button placement regarding the selector element, prepend, append
-            var position =  parts[1];
-            /////////////////////////////////////////////////////
-            // replace the selectors with bolt button identifiers
-            // if / when selectors are in the DOM
-            /////////////////////////////////////////////////////
-            onElementReady(identifier, function(element) {
-                if (getCheckoutKey() === '') return;
-                var bolt_button = document.createElement("div");
-                bolt_button.setAttribute("class", bolt_button_css_class);
-                // place the button before or after selector element
-                if (position && position.trim().toLowerCase() === 'append') {
-                    element.parentNode.insertBefore(bolt_button, element.nextSibling);
+        };
+        ////////////////////////////////////
+
+        /////////////////////////////////////////////////////
+        // Call Shipping and Tax prefetch & cache endpoint
+        /////////////////////////////////////////////////////
+        var prefetchShipping = function() {
+
+            if (!settings.prefetch_shipping) return;
+
+            var xhr;
+            xhr = window.XMLHttpRequest ? new XMLHttpRequest() : new ActiveXObject('Microsoft.XMLHTTP');
+
+            xhr.open('GET', settings.shipping_prefetch_url+'?orderReference='+cart.orderReference);
+            xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+
+            xhr.send();
+        };
+        /////////////////////////////////////////////////////
+
+        /////////////////////////////////////////////////////
+        // Create Bolt order and configure BoltCheckout
+        /////////////////////////////////////////////////////
+        var create_request;
+        var createOrder = function () {
+            if (create_request) create_request.abort();
+            // preset success condition to false
+            cart.orderToken = '';
+            // define the params sent with the request variable
+            var params = [];
+
+            // check and set payment_only flag
+            if (getCheckoutType() === 'payment') {
+
+                // stop if customer email field is present and invalid
```

Comment:
not sure if changes in this file is relevant for virtual items upport

Reply:
More than relevant. The file is pretty much refactored, some Magento objects are required. There is a *is guest checkout allowed* validation / login popup on bolt button click. Virtual / downloadable goods are allowed to logged in customers only on most setups.